pkg/handler/unrouted_handler.go

```diff
@@ -1337,7 +1366,19 @@ func (handler *UnroutedHandler) lockUpload(c *httpContext, id string) (Lock, err
 // isResumableUploadDraftRequest returns whether a HTTP request includes a sign that it is
 // related to resumable upload draft from IETF (instead of tus v1)
 func (handler UnroutedHandler) isResumableUploadDraftRequest(r *http.Request) bool {
-	return handler.config.EnableExperimentalProtocol && r.Header.Get("Upload-Draft-Interop-Version") == currentUploadDraftInteropVersion
+	interopVersionHeader := handler.getResumeableUploadDraftVersion(r)
+	supportedInteropVersion := interopVersionHeader == Version3 || interopVersionHeader == Version4
+	return handler.config.EnableExperimentalProtocol && supportedInteropVersion
+}
+
+func (handler UnroutedHandler) getResumeableUploadDraftVersion(r *http.Request) DraftVersion {
```

Comment:
```suggestion
func getResumableUploadDraftVersion(r *http.Request) DraftVersion {
```

This is not need to be a method on the UnroutedHandler struct.